vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImpl.java

```diff
@@ -442,6 +484,43 @@ private MultiMap getQueryParams() {
     return bodyEndHandlers;
   }
 
+  private Map<Integer, Handler<Void>> getExceptionHandlers() {
+    if (exceptionHandlers == null) {
+      // order is important we we should traverse backwards
+      exceptionHandlers = new TreeMap<>(Collections.reverseOrder());
+      response().exceptionHandler(v -> exceptionHandlers.values().forEach(handler -> handler.handle(null)));
+    }
+    return exceptionHandlers;
+  }
+
+  private Map<Integer, Handler<Void>> getCloseHandlers() {
+    if (closeHandlers == null) {
+      // order is important we we should traverse backwards
+      closeHandlers = new TreeMap<>(Collections.reverseOrder());
+      response().closeHandler(v -> closeHandlers.values().forEach(handler -> handler.handle(null)));
+    }
+    return closeHandlers;
+  }
+
+  private Map<Integer, Handler<Void>> getEndHandlers() {
+    if (endHandlers == null) {
+      // order is important we we should traverse backwards
+      endHandlers = new TreeMap<>(Collections.reverseOrder());
+
+      final Handler<Void> endHandler = v -> {
+        if (!endHandlerCalled) {
+          endHandlers.values().forEach(handler -> handler.handle(null));
```

Comment:
Should we wrap the for each lambda with a try catch, so, in case of a user error at the handler level, the remaining handlers also get called?